app/utils/local_drive_identity.py

```
@@ -0,0 +1,107 @@
+import subprocess
+from datetime import datetime, timezone
+from pathlib import Path
+from typing import Dict, Optional
+
+from app.models import ColdStorageBackendType, ColdStorageLocation
+
+
+def _diskutil_info(path: Path) -> Dict:
+    """Return diskutil plist info for a path on macOS, or empty dict if unavailable."""
+    try:
+        proc = subprocess.run(
```

Ruff (0.15.10)

[error] `subprocess` call: check for execution of untrusted input

(S603)